internal/attestation/crafter/crafter.go

```diff
@@ -517,26 +517,23 @@ func (c *Crafter) AddMaterialFromContract(ctx context.Context, attestationID, ke
 // AddMaterialContactFreeAutomatic adds a material to the crafting state checking the incoming material matches any of the
 // supported types in validation order. If the material is not found it will return an error.
 func (c *Crafter) AddMaterialContactFreeAutomatic(ctx context.Context, attestationID, value string, casBackend *casclient.CASBackend, runtimeAnnotations map[string]string) (schemaapi.CraftingSchema_Material_MaterialType, error) {
-	var kind schemaapi.CraftingSchema_Material_MaterialType
-	var found bool
-
-	// We want to run the material validation in a specific order
-	for _, kind = range schemaapi.CraftingMaterialInValidationOrder {
-		if err := c.AddMaterialContractFree(ctx, attestationID, kind.String(), value, casBackend, runtimeAnnotations); err != nil {
-			c.logger.Debug().Err(err).Str("kind", kind.String()).Msg("failed to add material")
-			continue
+	for _, kind := range schemaapi.CraftingMaterialInValidationOrder {
+		err := c.AddMaterialContractFree(ctx, attestationID, kind.String(), value, casBackend, runtimeAnnotations)
+		if err == nil {
+			// Successfully added material, return the kind
+			return kind, nil
 		}
-		// If we found a match we break the loop and stop looking
-		found = true
-		break
-	}
 
-	// Return an error if the material could not be added
-	if !found {
-		return kind, fmt.Errorf("failed to add material with attestationID: %s", attestationID)
+		c.logger.Debug().Err(err).Str("kind", kind.String()).Msg("failed to add material")
+
+		// Handle base error for upload and craft errors except the opening file error
+		if errors.Is(err, materials.ErrBaseUploadAndCraft) {
+			return kind, err
+		}
 	}
 
-	return kind, nil
+	// Return an error if no material could be added
+	return schemaapi.CraftingSchema_Material_MATERIAL_TYPE_UNSPECIFIED, fmt.Errorf("failed to add material with attestationID: %s", attestationID)
```

Comment:
why do you think `attestationID` is good info to be returned? It doesn't seem that valuable since the problem is not related to the attestationID but the passed value right? 

To me, a generic error would suffice ("failed to auto-adiscover material kind") or smth like that.